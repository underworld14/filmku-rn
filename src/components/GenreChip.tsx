import { View, ViewProps, Text } from "react-native";

type Props = ViewProps & {
  name: string;
};

export default function GenreChip({ name, ...props }: Props) {
  return (
    <View className="px-3 py-1 bg-blue-400 rounded-xl mr-1 mb-1" {...props}>
      <Text className="font-text-bold uppercase text-blue-700" style={{ fontSize: 9 }}>
        {name}
      </Text>
    </View>
  );
}
