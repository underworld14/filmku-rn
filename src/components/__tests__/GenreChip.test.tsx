import GenreChip from "components/GenreChip";
import renderer from "react-test-renderer";
// import { render, screen, fireEvent, waitFor } from "@testing-library/react-native";

describe("GenreChip", () => {
  it("renders correctly", () => {
    const tree: any = renderer.create(<GenreChip name="Action" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
