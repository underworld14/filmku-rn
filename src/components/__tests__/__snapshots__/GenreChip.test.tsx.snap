// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreChip renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingLeft": 12,
        "paddingRight": 12,
      },
      Object {
        "paddingBottom": 4,
        "paddingTop": 4,
      },
      Object {
        "backgroundColor": "#60a5fa",
      },
      Object {
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
      },
      Object {
        "marginRight": 4,
      },
      Object {
        "marginBottom": 4,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Mulish_700Bold",
        },
        Object {
          "textTransform": "uppercase",
        },
        Object {
          "color": "#1d4ed8",
        },
        Object {
          "fontSize": 9,
        },
      ]
    }
  >
    Action
  </Text>
</View>
`;
